src/main/java/org/broadinstitute/hellbender/tools/exome/orientationbiasvariantfilter/OrientationBiasFilterer.java

```diff
@@ -231,36 +243,31 @@ private static double calculateFob(final Genotype genotype, final boolean isRele
+                                                                                final Map<Transition, Long> transitionCount) {
         // Adjust the number to cut based on artifact mode
         final Map<Transition, Long> transitionNumToCut = new HashMap<>();
         transitionCount.keySet().stream().forEach(transition -> transitionNumToCut.put(transition, 0L));
         for (final Transition transition : transitionNumToCut.keySet()) {
-            transitionNumToCut.put(transition, Long.valueOf(Math.round(totalNumToCut * transitionCount.get(transition) / allTransitionCount)));
-            logger.info(sampleName + ": Cutting (" + transition + ") pre-preAdapterQ: " + transitionNumToCut.get(transition));
+            transitionNumToCut.put(transition, (long) calculateTotalNumToCut(fdrThresh, transitionCount.get(transition),
+                    transition, genotypesToConsiderForFiltering) );
+            logger.info(sampleName + ": Cutting (" + transition + ") pre-preAdapterQ: " + transitionNumToCut.get(transition) + " of " + transitionCount.get(transition));
         }
+
+        logger.info(sampleName + ": Cutting (total) pre-preAdapterQ: " + String.valueOf(transitionNumToCut.values().stream().mapToLong(Long::longValue).sum()));
+
         return transitionNumToCut;
     }
 
-    private static int calculateTotalNumToCut(final double fdrThresh, final long unfilteredGenotypeCount, final SortedMap<UniqueIDWrapper<Genotype>, VariantContext> genotypesToConsiderForFiltering) {
+    private static int calculateTotalNumToCut(final double fdrThresh, final long unfilteredGenotypeCount, final Transition transition,
+                                              final SortedMap<UniqueIDWrapper<Genotype>, VariantContext> genotypesToConsiderForFiltering) {
         final List<Double> pArtifactScores = genotypesToConsiderForFiltering.keySet().stream()
+                .filter(g -> OrientationBiasUtils.isGenotypeInTransition(g.getWrapped(), transition))
                 .map(g -> OrientationBiasUtils.getGenotypeDouble(g.getWrapped(), OrientationBiasFilterConstants.P_ARTIFACT_FIELD_NAME, 0.0))
                 .collect(Collectors.toList());
 
-        // When doing the Benjamini-Hochberg procedure, we need to include the non-artifact mode SNVs to guarantee the
-        //  FDR is kept to the specified threshold.
-        //   Pad the list of pArtifact scores so that the the non-artifacts (w/ a p-value of zero) are included.
-        final int numToPadZeroes = (int) unfilteredGenotypeCount - pArtifactScores.size();
-        List<Double> finalPArtifactScores = new ArrayList<>();
-        finalPArtifactScores.addAll(pArtifactScores);
-        finalPArtifactScores.addAll(Collections.nCopies(numToPadZeroes, 0.0));
-
-        return calculateTotalNumToCut(fdrThresh, unfilteredGenotypeCount, finalPArtifactScores);
+        return calculateTotalNumToCut(fdrThresh, pArtifactScores.size(), pArtifactScores);
```

Comment:
Technically, yes, but I left it separate in case we want to (effectively) zero pad.  I just added another method that will grab the .size() for this case.